ckanext/harvest/logic/action/update.py

```diff
@@ -586,11 +588,21 @@ def harvest_jobs_run(context, data_dict):
                     get_action('harvest_source_reindex')(
                         context, {'id': job_obj.source.id})
 
-                    status = get_action('harvest_source_show_status')(context, {'id': job_obj.source.id})
+                    status = get_action('harvest_source_show_status')(
+                        context, {'id': job_obj.source.id})
 
-                    if toolkit.asbool(config.get('ckan.harvest.status_mail.errored'))\
-                            and (status['last_job']['stats']['errored']):
-                        send_error_mail(context, job_obj.source.id, status)
+                    notify_all = toolkit.asbool(config.get('ckan.harvest.status_mail.all'))
+                    notify_errors = toolkit.asbool(config.get('ckan.harvest.status_mail.errored'))
+                    last_job_errors = status['last_job']['stats'].get('errored', 0)
+                    log.info('Notifications: All:{} On error:{} Errors:{}'.format(notify_all, notify_errors, last_job_errors))
+                    
+                    if notify_all:
```

Comment:
The duplicated `send_error_email` is a bit hard to follow. How about inverting the conditions to be about the error status vs success status? Something like:

```python
if last_job_errors > 0:  # completed with error
  if notify_all or notify_error:
    send_error_email(context, job_obj.source_id, status)
else: # completed succesfully
  if notify_all:
    send_summary_email(context, job_obj.source.id, status)
```

or (maybe too compact):

```python
if last_job_errors > 0 and (notify_all or notify_error):
  send_error_email(context, job_obj.source_id, status)
elif notify_all:
  send_summary_email(context, job_obj.source.id, status)
```